helpers/argumentParsers.go

```diff
@@ -1766,3 +1766,133 @@ func ParseSocketOption(rawValue uint64) (SocketOptionArgument, error) {
 	}
 	return v, nil
 }
+
+type BpfProgTypeArgument int32
```

Comment:
What is this different from the type BPFProgType already existent ? Can you re-use the type and functions we already have for bpf prog type ?

Reply:
If you check line 1128 and on, you will find we have bpf prog type, bpf map type, bpf attachment types, etc.

Reply:
thanks @rafaeldtinoco,

i didn't see that.

the `BPFProgType` is defined in `package libbpfgo`. 
i think that the parsing function i'm suggesting should be in `package helpers`. so, using this already existing type would create a problem of circular dependencies - `package helpers` would import `package libbpfgo` which imports `package helpers`.

how do you suggest i can use this existing type? 
move it to a new package maybe? or move it into `package helpers`?

i agree we shouldn't define same type with its funcs more than once.

Reply:
@roikol I think that helpers should be able to import libbpfgo but libbpfgo should _not_ import helpers. Before this PR it was only imported for the RWArray (line 1504). 

I think RWArray should be moved over to the libbpfgo package, or to it's own package. That way you don't have to redefine all the same constants again here, and you can cleanly have the parse functions in helpers.

I will open a PR momentarily with this fix, once that's merged you can update this PR.

Reply:
thank you @grantseltzer !

sounds good. wasn't really sure what's the best approach here.
so i'll wait for your PR to be merged and then i'll update this one.

thanks :)

Reply:
Take a look at #202  when you get a chance

Reply:
Merged 202, you can rebase your changes on top of that